src/ResourceManager/Sql/Commands.Sql/Database Backup/Cmdlet/AzureSqlDatabaseBackupShortTermRetentionPolicyCmdletBase.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Azure.Commands.Sql.Backup.Services;
+using Microsoft.Azure.Commands.Sql.Common;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.Sql.Database_Backup.Model;
+using Microsoft.Azure.Commands.Sql.Database.Model;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Sql.Database_Backup.Cmdlet
+{
+    public abstract class AzureSqlDatabaseBackupShortTermRetentionPolicyCmdletBase : AzureSqlCmdletBase<IEnumerable<AzureSqlDatabaseBackupShortTermRetentionPolicyModel>, AzureSqlDatabaseBackupAdapter>
+    {
+        /// <summary>
+        /// The expected number of segments in a short term retention policy resource id.
+        /// </summary>
+        private const int BackupShortTermRetentionPolicyResourceIdSegmentsLength = 12;
+
+        /// <summary>
+        /// Parameter set with ResourceGroup name, Server name and Database name.
+        /// </summary>
+        protected const string PolicyByResourceServerDatabaseSet = "PolicyByResourceServerDatabase";
+
+        /// <summary>
+        /// Parameter set for using a Database Input Object.
+        /// </summary>
+        private const string PolicyByInputObjectSet = "PolicyByInputObject";
+
+        /// <summary>
+        /// Parameter set for using a resource Id.
+        /// </summary>
+        private const string PolicyByResourceIdSet = "PolicyByResourceIdSet";
+
+        /// <summary>
+        /// Gets or sets the Database object to get the policy for.
+        /// </summary>
+        [Parameter(
+            ParameterSetName = PolicyByInputObjectSet,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "The database object to get the policy for.")]
+        [ValidateNotNullOrEmpty]
+        public AzureSqlDatabaseModel InputObject { get; set; }
+
```

Comment:
Usually the InputObject name is uses for parameter of the same type as the cmdlet output type. 
In this case you may want to rename the parameter from InputObject to AzureSqlDatabase, for example.

Reply:
Sure